### ISSUE TITLE
Host tempo sync

### ISSUE THREAD

**Issue by unfa** [zynaddsubfx/zyn-fusion-issues#239]
It's very common for synthesizers to get tempo information from the host and allow users to set their LFO frequencies, Delay times, and other stuff in musical values, as opposed to absolute time in seconds.

I think this is the main reason I'm not really using LFOs and Delays in Zyn for musical effects, and instead add these effects later in the processing chain with other stuff.

This could go even further - making it possible to have envelope times defined in musical values aswell, making portamento times definable in musical time etc.

I think that even basic BPM sync would be something that'd boost Zyn-Fusion's usability *a lot*.

What do you think?

**Comment by apoorv569** [comment]
> It's very common for synthesizers to get tempo information from the host and allow users to set their LFO frequencies, Delay times, and other stuff in musical values, as opposed to absolute time in seconds.
> 
> I think this is the main reason I'm not really using LFOs and Delays in Zyn for musical effects, and instead add these effects later in the processing chain with other stuff.
> 
> This could go even further - making it possible to have envelope times defined in musical values aswell, making portamento times definable in musical time etc.
> 
> I think that even basic BPM sync would be something that'd boost Zyn-Fusion's usability _a lot_.
> 
> What do you think?

I was making a track recently and actually faced this issue, there is no option to sync LFO and envelops to the song/host tempo. It would be a great to have such a feature included.

**Comment by fundamental** [comment]
Odd, I thought this was already a documented enhancement request, but the only spot I'm seeing it in this repo looks like the GSoC 2019 ideas list. I'd agree this functionality would be welcome and I have some ideas as to how it would be implemented.

**Comment by falkTX** [comment]
Note that DPF can already supply this information, and the transport API is similar to the JACK one, so if we do it for the jack standalone, having it on the plugin side should be mostly copy&paste.

**Comment by mxmilkiib** [comment]
https://sourceforge.net/p/zynaddsubfx/feature-requests/36 and https://sourceforge.net/p/zynaddsubfx/feature-requests/76

**Comment by fundamental** [comment]
Yep, looks like I've already got references to the TimePosition information that DPF provides in the GSoC 2019 entry.

**Comment by friedolino78** [comment]
Now that the Global Tempo Parameter is in place this should be easy to add.
I originally planned to use Midi clock and Timecode Messages as Input.

**Comment by friedolino78** [comment]
with the host sync capabilities added for the reverse effect it should be even easier. with a litle bit of luck it is only a few code lines
